NumLib/Assembler/VectorMatrixAssembler.h

```diff
@@ -95,17 +95,14 @@ class VectorMatrixAssembler<GlobalMatrix, GlobalVector, LocalAssembler,
         const double t, GlobalVector const& x,
         GlobalMatrix& M, GlobalMatrix& K, GlobalVector& b) const
     {
-        auto cb = [&local_assembler](
-                std::vector<double> const& local_x,
-                LocalToGlobalIndexMap::RowColumnIndices const& r_c_indices,
-                const double t,
-                GlobalMatrix& M, GlobalMatrix& K, GlobalVector& b)
-        {
-            local_assembler.assemble(t, local_x);
-            local_assembler.addToGlobal(r_c_indices, M, K, b);
-        };
-
-        passLocalVector_(cb, id, _data_pos, x, t, M, K, b);
+        auto const x_and_indices = prepareLocalVector(id, _data_pos, x);
+        auto const& local_x = std::get<0>(x_and_indices);
```

Comment:
I think there is not much benefit from using `std::tie` here: prepareLocalVector() has to return by value, the involved copy might be optimized out by the compiler. `local_x` and `indices` are only references, they will probably be optimized out anyway.